planner/core/planbuilder.go

```diff
@@ -855,7 +859,10 @@ func (b *PlanBuilder) buildAnalyzeTable(as *ast.AnalyzeTableStmt) (Plan, error)
 }
 
 func (b *PlanBuilder) buildAnalyzeIndex(as *ast.AnalyzeTableStmt) (Plan, error) {
-	p := &Analyze{MaxNumBuckets: as.MaxNumBuckets}
+	p := &Analyze{}
+	if err := handleAnalyzeOptions(p, as.AnalyzeOpts); err != nil {
```

Comment:
What about move `handleAnalyzeOptions` to `buildAnalyze`. Then we can reduce calculation and memory cost.